### PR TITLE
[matter_yamltests] Add CommissionerCommands and DiscoveryCommands method definitions

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem

`CommissionerCommands` and `DiscoveryCommands` does not have a definition. It makes errors happening at runtime instead of parsing time.

### CODE REVIEW

**Review by tehampson — APPROVED** [review]

File: scripts/py_matter_yamltests/matter_yamltests/pseudo_clusters/clusters/commissioner_commands.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#
```
[the remark this answers is not in the document]
Comment (vivien-apple) [reply scripts/py_matter_yamltests/matter_yamltests/pseudo_clusters/clusters/commissioner_commands.py:1]: If the cluster does not have a definition #25109 should not trigger an error.

File: scripts/py_matter_yamltests/matter_yamltests/pseudo_clusters/clusters/discovery_commands.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    <command source="client" code="3" name="FindCommissionableByCommissioningMode" response="FindResponse">
+    </command>
+
+    <command source="client" code="4" name="FindCommissionableByVendorId" response="FindResponse">
+        <arg name="value" type="vendor_id"/>
+    </command>
+
+    <command source="client" code="5" name="FindCommissionableByDeviceType" response="FindResponse">
+        <arg name="value" type="int16u"/>
```
Comment (bzbarsky-apple) [inline scripts/py_matter_yamltests/matter_yamltests/pseudo_clusters/clusters/discovery_commands.py:43]: Any reason this is not `devtype_id`?

File: scripts/py_matter_yamltests/matter_yamltests/pseudo_clusters/clusters/discovery_commands.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    <command source="client" code="6" name="FindCommissioner" response="FindCommissionerResponse">
+    </command>
+
+    <command source="client" code="7" name="FindCommissionerByVendorId" response="FindCommissionerResponse">
+        <arg name="value" type="vendor_id"/>
+    </command>
+
+    <command source="client" code="8" name="FindCommissionerByDeviceType" response="FindCommissionerResponse">
+        <arg name="value" type="int16u"/>
```
Comment (bzbarsky-apple) [inline scripts/py_matter_yamltests/matter_yamltests/pseudo_clusters/clusters/discovery_commands.py:54]: Again, `devtype_id`?

File: scripts/py_matter_yamltests/matter_yamltests/pseudo_clusters/clusters/discovery_commands.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+    <command source="server" code="9" name="FindResponse">
+        <arg name="hostName" type="char_string"/>
+        <arg name="instanceName" type="char_string"/>
+        <arg name="longDiscriminator" type="int16u"/>
+        <arg name="shortDiscriminator" type="int16u"/>
+        <arg name="vendorId" type="vendor_id"/>
+        <arg name="productId" type="int16u"/>
+        <arg name="commissioningMode" type="int8u"/>
+        <arg name="deviceType" type="int16u"/>
```
Comment (bzbarsky-apple) [inline scripts/py_matter_yamltests/matter_yamltests/pseudo_clusters/clusters/discovery_commands.py:65]: devtype_id?

File: scripts/py_matter_yamltests/matter_yamltests/pseudo_clusters/clusters/discovery_commands.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        <arg name="deviceName" type="char_string"/>
+        <arg name="rotatingId" type="octet_string"/>
+        <arg name="rotatingIdLen" type="int64u"/>
```
Comment (bzbarsky-apple) [inline scripts/py_matter_yamltests/matter_yamltests/pseudo_clusters/clusters/discovery_commands.py:68]: Why do we have the len, given that octet_string knows its own length?  Followup to make this make more sense?

**Review by bzbarsky-apple — APPROVED** [review]

File: scripts/py_matter_yamltests/matter_yamltests/pseudo_clusters/clusters/discovery_commands.py (hunk 5 above)
Comment (vivien-apple) [reply scripts/py_matter_yamltests/matter_yamltests/pseudo_clusters/clusters/discovery_commands.py:68]: Well, the previous API has it. Changing it now means a ZAP update. I can't remember why I have added it initially.

Comment (vivien-apple) [reply scripts/py_matter_yamltests/matter_yamltests/pseudo_clusters/clusters/discovery_commands.py:68]: https://github.com/project-chip/connectedhomeip/issues/25115

**Pull request merged**